checker/src/main/java/org/checkerframework/checker/mustcall/MustCallVisitor.java

```diff
@@ -0,0 +1,193 @@
+    // precisely?
+    if (!checker.hasOption(MustCallChecker.NO_LIGHTWEIGHT_OWNERSHIP) && enclosingMethod != null) {
+      ExecutableElement methodElt = TreeUtils.elementFromDeclaration(enclosingMethod);
+      AnnotationMirror notOwningAnno = atypeFactory.getDeclAnnotation(methodElt, NotOwning.class);
+      if (notOwningAnno != null) {
+        // skip return type subtyping check, because not-owning pointer means OCC won't check anyway
+        return null;
+      }
+    }
+    return super.visitReturn(node, p);
+  }
+
+  @Override
+  protected boolean skipReceiverSubtypeCheck(
+      MethodInvocationTree node,
+      AnnotatedTypeMirror methodDefinitionReceiver,
+      AnnotatedTypeMirror methodCallReceiver) {
+    // TODO: Check explicit receiver parameters annotated with @Owning. ExecutableElement
+    //       doesn't have any way to get an element associated with the receiver, so I can't
+    //       figure out a way to get a declaration annotation for the receiver. It might not
+    //       be possible? The below is the closest that I got, but the receiver doesn't show up
+    //       in the list of the parameters, even when it's explicit. Is this a bug in javac?
+    //
+    //    ExecutableElement elt = TreeUtils.elementFromUse(node);
+    //    System.out.println(elt);
+    //    List<? extends VariableElement> params = elt.getParameters();
+    //    if (!params.isEmpty()) {
+    //      VariableElement first = params.get(0);
+    //      if (first.getSimpleName().contentEquals("this")) {
+    //        return atypeFactory.getDeclAnnotation(first, Owning.class) == null;
+    //      }
+    //    }
+    return true;
+  }
+
+  /**
+   * Mark (using the extraArgs) any assigments where the LHS is a resource variable, so that close
+   * doesn't need to be considered.
+   */
+  @Override
+  protected void commonAssignmentCheck(
+      Tree varTree,
+      ExpressionTree valueExp,
+      @CompilerMessageKey String errorKey,
+      Object... extraArgs) {
+    if (TreeUtils.elementFromTree(varTree).getKind() == ElementKind.RESOURCE_VARIABLE) {
+      // Use the extraArgs array to signal to later stages of the CAC that this is in a
+      // resource variable context.
+      Object[] newExtraArgs = Arrays.copyOf(extraArgs, extraArgs.length + 1);
+      newExtraArgs[newExtraArgs.length - 1] = ElementKind.RESOURCE_VARIABLE;
+      super.commonAssignmentCheck(varTree, valueExp, errorKey, newExtraArgs);
+    } else {
+      super.commonAssignmentCheck(varTree, valueExp, errorKey, extraArgs);
+    }
+  }
+
+  /**
+   * If the LHS has been marked as a resource variable, then the standard CAC is skipped and a check
+   * that does not include "close" is substituted.
+   */
+  @Override
+  protected void commonAssignmentCheck(
+      AnnotatedTypeMirror varType,
+      AnnotatedTypeMirror valueType,
+      Tree valueTree,
+      @CompilerMessageKey String errorKey,
+      Object... extraArgs) {
```

Comment:
There was already a note on `commonAssignmentCheck(Tree, ExpressionTree, String, Object...)`, but I've added one here as well and expanded that note so that the two reference each other.